Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,520 @@ public override bool IsServerConnected()
+                    foreach (QC.QCTSTestStep step in tc.Steps)
+                    {
+                        Activity stepActivity;
+                        bool toAddStepActivity = false;
+
+                        //check if mapped activity exist in repository
+                        Activity repoStepActivity = (Activity)GingerActivitiesRepo.Where(x => x.ExternalID == step.StepID).FirstOrDefault();
+                        if (repoStepActivity != null)
+                        {
+                            //check if it is part of the Activities Group
+                            ActivityIdentifiers groupStepActivityIdent = (ActivityIdentifiers)tcActivsGroup.ActivitiesIdentifiers.Where(x => x.ActivityExternalID == step.StepID).FirstOrDefault();
+                            if (groupStepActivityIdent != null)
+                            {
+                                //already in Activities Group so get link to it
+                                stepActivity = (Activity)busFlow.Activities.Where(x => x.Guid == groupStepActivityIdent.ActivityGuid).FirstOrDefault();
+                                // in any case update description/expected/name - even if "step" was taken from repository
+                                stepActivity.Description = step.Description;
+                                stepActivity.Expected = step.Expected;
+                                stepActivity.ActivityName = tc.TestName + ">" + step.StepName;
+                            }
+                            else//not in ActivitiesGroup so get instance from repo
+                            {
+                                stepActivity = (Activity)repoStepActivity.CreateInstance();
+                                toAddStepActivity = true;
+                            }
+                        }
+                        else//Step not exist in Ginger repository so create new one
+                        {
+                            stepActivity = new Activity();
+                            stepActivity.ActivityName = tc.TestName + ">" + step.StepName;
+                            stepActivity.ExternalID = step.StepID;
+                            stepActivity.Description = step.Description;
+                            stepActivity.Expected = step.Expected;
+
+                            toAddStepActivity = true;
+                        }
+
+                        if (toAddStepActivity)
+                        {
+                            //not in group- need to add it
+                            busFlow.AddActivity(stepActivity, tcActivsGroup);
+                        }
+
+                        //pull TC-Step parameters and add them to the Activity level
+                        List<string> stepParamsList = new List<string>();
+                        stepParamsList = GetTCParameterList(step.Description);
+                        //GetStepParameters(StripHTML(step.Expected), ref stepParamsList);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)